src/lib/mdns/minimal/records/Txt.h

```diff
@@ -0,0 +1,68 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <string.h>
+
+#include <mdns/minimal/records/ResourceRecord.h>
+
+namespace mdns {
+namespace Minimal {
+
+class TxtResourceRecord : public ResourceRecord
+{
+public:
+    static constexpr uint64_t kDefaultTtl = 10;
```

Comment:
The TTL here should not be shorter than the overall entry TTL if possible (unless I misundertand the TTL here)

Reply:
Every record has its own TTL. I would defer TTL handling to the future. Actually the APP should set these and never rely on defaults. It is just that I was lazy in the example so I have set a default here.